### DISCUSSION TITLE
Filter method without filter parameters doesn't make sense (clarification)

### DISCUSSION THREAD

**Discussion by JakenVeina** [reactivemarbles/DynamicData#734] [Q&A]
Can I ask for some clarification on what you said in https://github.com/reactivemarbles/DynamicData/issues/219? I had the same question and found the issue, but I don't see how the answer explains anything about the question.

> Reapply filter is an option which can be used when filtering on mutable values.

But WHAT is the filtering when using this overload, that doesn't specify a filter?

> I recommend you down load this [DynamicData Snippets](https://github.com/RolandPheasant/DynamicData.Snippets) where there are some examples of dd filtering + unit tests to illustrate expectations

There's no examples in that project that demonstrate using this overload of `.Filter()`, with only a `reapplyFilter` parameter. Not that I could find.

I thought maybe this might be referring to filtering applied when you use `.Connect()` with the overload that takes a filter parameter, so I tried it out, and `reapplyFilter` doesn't cause the filtering to reapply. Same thing if I just use an adjacent call to `.Filter()` either before or after the one with `reapplyFilter`.

```cs
public record ResourceVersionEntity
{
    public required ulong Id { get; init; }

    public required ulong ResourceId { get; init; }

    public ulong? PreviousId { get; init; }

    public required string Name { get; init; }
}

var resourceVersions = new SourceCache<ResourceVersionEntity, ulong>(resourceVersion => resourceVersion.Id);

var previousIdIndex = resourceVersions
    .Connect(resourceVersion => resourceVersion.PreviousId.HasValue)
    .Group(resourceVersion => resourceVersion.PreviousId!.Value)
    .AsObservableCache();

var currentResourceVersions = resourceVersions
    .Connect(resourceVersion => !previousIdIndex.Lookup(resourceVersion.Id).HasValue)
    .Filter(previousIdIndex.Connect().Select(static _ => Unit.Default))
    .ChangeKey(resourceVersion => resourceVersion.ResourceId)
    .AsObservableCache();

resourceVersions.AddOrUpdate(new ResourceVersionEntity()
{
    Id          = 4,
    ResourceId  = 2,
    PreviousId  = 2,
    Name        = "My Renamed Resource"
});

resourceVersions.AddOrUpdate(new ResourceVersionEntity()
{
    Id          = 1,
    ResourceId  = 1,
    Name        = "My Resource"
});

resourceVersions.AddOrUpdate(new ResourceVersionEntity()
{
    Id          = 2,
    ResourceId  = 2,
    Name        = "My Other Resource"
});

resourceVersions.AddOrUpdate(new ResourceVersionEntity()
{
    Id          = 3,
    ResourceId  = 3,
    Name        = "My Additional Resource"
});

resourceVersions.AddOrUpdate(new ResourceVersionEntity()
{
    Id          = 5,
    ResourceId  = 2,
    PreviousId  = 4,
    Name        = "My Restored Resource"
});
```

**Comment by JakenVeina** [comment]
I probably should have just done this in the first place. Looking at the source, this operator can literally never emit any items, no matter what parameters you give it.

This overload was originally added by 3657fee1, where @RolandPheasant was doing a large scale re-organization, which included consolidating code for all the different `.Filter()` overloads into the shared `StaticFilter` and `DynamicFilter` internal classes, and it looks like he just took an effort to add overloads for all different combinations of optional parameters, without considering that `predicateChanged` isn't actually an optional parameter.

Seems like this overload should probably just be removed.

**Reply by RolandPheasant** [reply]
I have to be honest, I really cannot remember the details or the thought process! However having an overload without params is almost certainly a mistake / oversight.

**Reply by dwcullop** [reply]
Yeah, I can't imagine what that would mean either.
